BSP/STM32F413/Src/BSP_Contactor.c

```diff
@@ -45,20 +123,32 @@ void BSP_Contactor_On(void) {
  * @param   None
  * @return  None
  */
-void BSP_Contactor_Off(void) {
-    GPIO_WriteBit(GPIOB, GPIO_Pin_0, Bit_RESET);	// Set pin 0 low
+void BSP_Contactor_Off(CONT_CHOICE contactorChoice) {
+    // set output pins LOW
+	if (contactorChoice & ARRAY_CONTACTOR) {
+		GPIO_WriteBit(C1_PORT, GPIO_Pin_0, Bit_RESET);
+	}
+	if (contactorChoice & HVHIGH_CONTACTOR) {
+		GPIO_WriteBit(C2_PORT, GPIO_Pin_4, Bit_RESET);
+	}
+	if (contactorChoice & HVLOW_CONTACTOR) {
+		GPIO_WriteBit(C3_PORT, GPIO_Pin_0, Bit_RESET);
+	}
 }
 
 /**
  * @brief   Gets the state of the Contactor switch from one of its AUX pins.
  * @param   None
  * @return  0 if contactor is off/open, 1 if on/closed
  */
-bool BSP_Contactor_GetState(void) {
-    // The AUX pin we have connected to is negative logic
-    if((GPIOB->IDR & GPIO_Pin_1) >> 1) {
-		return 0;
-	} else {
-		return 1;
+bool BSP_Contactor_GetState(CONT_CHOICE contactorChoice) {
+	if (contactorChoice & ARRAY_CONTACTOR) {
+		return ((C1_PORT->IDR & GPIO_Pin_1) >> 1) ? 0 : 1;
```

Comment:
also, just a thought for more complex functions it's usually best practice to have a single return path, i.e.
```
bool return_value;
if (...) {
   return_value = ...
}
else if (...) {
    return_value = ...
}
return return_value
```
not really necessary to change it here, just my 2¢
(also I write multiple return paths all the time...)